iModelCore/ecobjects/src/SchemaMerger.cpp

```diff
@@ -237,8 +239,11 @@ BentleyStatus SchemaMerger::MergeSchemas(SchemaMergeResult& result, bvector<ECSc
                     result.Issues().ReportV(IssueSeverity::Fatal, IssueCategory::BusinessProperties, IssueType::ECSchema, "Schema '%s' from %s side failed to be copied.", schema->GetFullSchemaName().c_str(), side);
                     failedToFillSchemas = true;
                     }
-                copiedSchema->SetOriginalECXmlVersion(schema->GetOriginalECXmlVersionMajor(), schema->GetOriginalECXmlVersionMinor());
-                result.GetSchemaCache().AddSchema(*copiedSchema);
+                if (copiedSchema.IsValid())

```

Comment:
Nice catch, but better reverse this check, log an error and return ERROR in case the copied schema is not valid.

Reply:
Also, can we hit this with a test?

Reply:
The thing is, we're already doing that after we return from the lambda function.
We're checking if "failedToFillSchemas" got set to true and if it has, we're returning Error.
Also the error is being logged on line 239.
So, I just added the check.

And yes it can be hit from a test if the xml hasn't been read correctly. For eg, invalid schema name etc